### PR TITLE
chore(runtime): set exports field in client runtime package.json

### LINKED ISSUE #5275: bug: Stencil is generating invalid ESM COde
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.10.0

### Current Behavior

Im having a tsx file with the following import:

      import { Component, Prop, h, State } from '@stencil/core';

and stencil when building a custom-element is creating this import:
 
      import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

but this is invalid.

### Expected Behavior

I try to use stencil s a ESM module with an importmap, so the importmap is created from the packages.json export directive, stencil has none, so the import should look like this:

      import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client/index.js';

or stay in this version:

      import { proxyCustomElement, HTMLElement, h } from '@stencil/core';

or the package.json should contain an export directive like this:

    "exports": {
        ".": "./index.js",
        "./internal/client": {
          "default": "./internal/client/index.js"
        }
      }
    

     

### System Info

_No response_

### Steps to Reproduce

Build a library with stencil

### Code Reproduction URL

--

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by christian-bromann**
## What is the current behavior?
fixes #5275

## What is the new behavior?
Set an `exports` field in the `package.json` for the client runtime so it can be imported by ESM.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

n/a

## Other information

n/a

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,34 @@
...
+    },
+    "./internal/testing": {
+      "import": "./internal/testing/index.js"
+    },
+    "./internal/testing/*": {
+      "import": "./internal/testing/*"
+    },
+    "./internal/app-data": {
+      "import": "./internal/app-data/index.js",
+      "require": "./internal/app-data/index.cjs"
```
Comment (tanner-reits) [inline package.json:44]: Still think it's so weird we expose things under "internal" 😅

**Review by tanner-reits — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (christian-bromann) [reply package.json:44]: Totally! These aren't user interfaces though , they are all just import references that Stencil is using.

Comment (jogibear9988) [reply package.json:44]: you could remove them, but then your imports should use relative paths, with js extension

Comment (christian-bromann) [reply package.json:44]: Yep, we will do more optimizations in this direction, e.g. move to ESM

**Pull request merged**